k9mail/src/main/java/com/fsck/k9/ui/compose/QuotedMessagePresenter.java

```diff
@@ -173,7 +174,8 @@ public void onSaveInstanceState(Bundle outState) {
     public void onRestoreInstanceState(Bundle savedInstanceState) {
         quotedHtmlContent = (InsertableHtmlContent) savedInstanceState.getSerializable(STATE_KEY_HTML_QUOTE);
         if (quotedHtmlContent != null && quotedHtmlContent.getQuotedContent() != null) {
-            view.setQuotedHtml(quotedHtmlContent.getQuotedContent());
+            // we don't have the part here, but inline-displayed images are cached by the webview
+            view.setQuotedHtml(quotedHtmlContent.getQuotedContent(), null);
```

Comment:
This feels very hacky. You could save the mapping and recreate a `AttachmentResourceProvider` instance from it. Or make `AttachmentResourceProvider` implement `Parcelable`.